src/manager/TickManager.ts

```diff
@@ -0,0 +1,46 @@
+import BN from "bn.js";
+import { Tick } from "../model/Tick";
+import { jsonMapMember, jsonObject } from "typedjson";
+
+@jsonObject
+export class TickManager {
+  @jsonMapMember(Number, Tick, { name: "ticks_json" })
+  private ticks: Map<number, Tick>;
+
+  constructor(ticks: Map<number, Tick> = new Map()) {
+    this.ticks = ticks;
+  }
+
+  get(tickIndex: number): Tick {
+    return this.ticks.get(tickIndex) || new Tick();
+  }
+
+  set(tickIndex: number, tick: Tick) {
+    this.ticks.set(tickIndex, tick);
+  }
+
+  flipTick(tickIndex: number, tickSpacing: number) {
+    // TODO
+  }
+
+  nextInitializedTick(
```

Comment:
Use verb in all functions, maybe pre-pend `add`.

Reply:
I see and I think `pick` as pre-pend is better here.

Reply:
I meant to say `get`